### PR TITLE
[TTIR] Add SDPA fusing at TTIR level

### PR DESCRIPTION
**Pull request by jserbedzijaTT**
## Summary

Adds an SDPA fusing pattern to `TTIRFusingPass` that matches the canonical mathematical form `matmul(softmax((Q @ Kᵀ) * scale + mask), V)` into a single `ttir.scaled_dot_product_attention`. Matching at TTIR (before TM-reordering) keeps the matcher close to the math; the TTNN-level matcher stays as fallback. Migration unblocked by #8596.

The matcher also covers the forms real models lower to: Kᵀ as last-two-dims `ttir.permute` (dot_general form), scale on Q/K/post-score (mul or div, `ttir.full` const), GQA `repeat_interleave` peeled to native Hkv, NaN-safe `where(cond, 0, softmax)` re-applied post-op, attention sink (concat sink column / slice off) into `attention_sink`, softmax precision casts. Strict 4D, last-dim softmax; declines anything else.

Supporting changes:
- **TTIRToTTNN**: thread `attention_sink` through the Sq=1 decode lowering; compensate sink by 1/scale (kernel scales the sink — tt-metal#40470), keeping TTIR op + golden faithful.
- **TTNN decode op**: add `sliding_window_size` (td, flatbuffer, serializer, runtime); decode→prefill decomposition bakes the window into an additive mask anchored at cur_pos / Sk-1.
- **`nlp_concat_heads_decode` fusing**: handle the canonicalized reshape reorder, typecasts, and NaN-scrub where, with full rollback when validation declines.

Tests: 26 TTIR fusing lit cases (incl. 7 rejection guards), decode decomposition + silicon sink/sliding-window tests.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
### tt-xla Validation

**Base (uplifted):** `260d4c49`

| Test Suite | Run | Status |
|------------|-----|--------|
| full-mlir-uplift-qualification | [Run #29571190828](https://github.com/tenstorrent/tt-xla/actions/runs/29571190828) | :white_check_mark: passed |

### CODE REVIEW

**Review by sgholamiTT — APPROVED** [review]

File: test/ttmlir/Dialect/TTNN/optimizer/ttnn_fusing/nlp_concat_heads_decode/nlp_concat_heads_decode.mlir [hunk 1]
```diff
@@ -6,6 +6,9 @@
 //   1. Basic decode pattern: permute + reshape fuses to nlp_concat_heads_decode
```
Comment (sdjordjevicTT) [inline test/ttmlir/Dialect/TTNN/optimizer/ttnn_fusing/nlp_concat_heads_decode/nlp_concat_heads_decode.mlir:6]: Can we also add builder test similar to what we added here:
https://github.com/tenstorrent/tt-mlir/blob/main/test/python/golden/ttir_ops/fusing/test_rope_fusing.py

**Review by sdjordjevicTT — APPROVED** [review]
Great! Just before merging, please run XLA benchmark tests to see if we may have missed some edge cases :)

**Review by jgrimTT — APPROVED** [review]
Tools changes LG

**Review by pilkicTT — APPROVED** [review]
